src/topology/homotopy.lean

```diff
@@ -0,0 +1,150 @@
+/-
+Copyright (c) 2021 Shing Tak Lam. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Shing Tak Lam
+-/
+
+import topology.unit_interval
+import topology.algebra.ordered.proj_Icc
+import topology.continuous_function.basic
+import topology.compact_open
+
+/-!
+# Homotopy between functions
+
+In this file, we define a `homotopy` between two functions `f₀` and `f₁`.
+
+## Definitions
+
+* `homotopy f₀ f₁` is the type of homotopies between `f₀` and `f₁`
+* `homotopy.refl f₀` is the constant homotopy between `f₀` and `f₀`
+* `homotopy.symm f₀ f₁` is a `homotopy f₁ f₀` defined by reversing the homotopy
+* `homotopy.trans F G`, where `F : homotopy f₀ f₁`, `G : homotopy f₁ f₂` is a `homotopy f₀ f₂`
+  defined by putting the first homotopy on `[0, 1/2]` and the second on `[1/2, 1]`.
+-/
+
+noncomputable theory
+
+universes u v
+
+variables {X : Type u} {Y : Type v} [topological_space X] [topological_space Y]
+
+open_locale unit_interval
+
+namespace continuous_map
+
+/--
+The type of homotopies between two functions.
+-/
+structure homotopy (f₀ f₁ : X → Y) extends C(I × X, Y) :=
```

Comment:
The definition itself does not require `f₀ f₁ : X → Y` to be continuous. However it is implied/required. Also, this is in the `continuous_map` namespace, so if we changed it to `C(X, Y)`, then dot notation would work. But is `f.homotopy g` something we want?

Reply:
Certainly `f.homotopic g` sounds very useful. And `f.homotopy g` doesn't seem harmful either.

Reply:
Alright, I've change it to take `C(X, Y)`